doc/endpoint/xml.md

```diff
@@ -0,0 +1,98 @@
+This would enable encoding objects to XML strings, and decoding XML strings to objects.
+Implementation is fairly easy, and for now, one guide on how to integrate with scalaxb is provided.
+
+```eval_rst
+.. note::
+
+  Note, that implementing `XmlCodec[T]` would require deriving not only XML library encoders/decoders, 
+  but also tapir related `Schema[T]`. These are completely separate - any customization e.g. for field
+  naming or inheritance strategies must be done separately for both derivations.
+  For more details see sections on [schema derivation](schemas.md) and on supporting [custom types](customtypes.md) in general.
+```
+
+## Scalaxb
+
+If you possess the XML Schema definition file (`.xsd` file) consider using the scalaxb tool,
+which would generate needed models and serialization/deserialization logic.
+To use the tool please follow the documentation on [setting up](https://scalaxb.org/setup) and
+[running](https://scalaxb.org/running-scalaxb) scalaxb.
+
+After code generation, create the `TapirXmlScalaxb` trait (or trait with another name of your choosing) and add the following code snippet:
+
+```scala
+import generated.`package`.defaultScope // import may differ depending on location of generated code
+import scalaxb.XMLFormat // import may differ depending on location of generated code
+import scalaxb.`package`.{fromXML, toXML} // import may differ depending on location of generated code
+import sttp.tapir.Codec.XmlCodec
+import sttp.tapir.DecodeResult.{Error, Value}
+import sttp.tapir.{Codec, EndpointIO, Schema, stringBodyUtf8AnyFormat}
+
+import scala.xml.{NodeSeq, XML}
+
+trait TapirXmlScalaxb {
+  case class XmlElementLabel(label: String)
+
+  def xmlBody[T: XMLFormat: Schema](implicit l: XmlElementLabel): EndpointIO.Body[String, T] = stringBodyUtf8AnyFormat(scalaxbCodec[T])
+
+  implicit def scalaxbCodec[T: XMLFormat: Schema](implicit label: XmlElementLabel): XmlCodec[T] = {
+    Codec.xml((s: String) =>
+      try {
+        Value(fromXML[T](XML.loadString(s)))
+      } catch {
+        case e: Exception => Error(s, e)
+      }
+    )((t: T) => {
+      val nodeSeq: NodeSeq = toXML[T](obj = t, elementLabel = label.label, scope = defaultScope)
+      nodeSeq.toString()
+    })
+  }
+}
+```
+This creates `XmlCodec[T]` that would encode / decode the types with `XMLFormat`, `Schema` and with `XmlElementLabel` provided in scope.
+It also introduces `xmlBody` helper method, which allows you to easily express, that the declared endpoint consumes or returns XML.
+
+
+Next to this trait, you might want to introduce `xml` package object to simplify imports.
+```scala
+package object xml extends TapirXmlScalaxb
+```
+
+From now on, XML serialization/deserialization would work for all classes generated from `.xsd` file as long as `XMLFormat`
+and `Schema` for the type in the question, `XmlElementLabel` for the top XML node would be implicitly provided in the scope.
```

Comment:
I think this could be a list, plus I would elaborate a bit on what's the label for

Reply:
updated